test/shared/src/main/scala-3/zio/test/Macros.scala

```diff
@@ -126,6 +126,12 @@ object SmartAssertMacros {
 
   def transform[A: Type](expr: Expr[A])(using PositionContext, Quotes) : Expr[TestArrow[Any, A]] = {
     import quotes.reflect._
+    def isBool(term: quotes.reflect.Term): Boolean = {
+      term.tpe.widen.asType match {
+        case '[Boolean] => true
```

Comment:
I don't like this function, but I have no idea if we can match just particular types in :
```
      case Unseal(MethodCall(lhs, "&&", tpes, Some(List(rhs)))) if isBool(lhs) =>
 ```